dashboard/rules.py

```diff
@@ -60,12 +67,13 @@ def is_instructor_in_course_id(self, user, course_id):
     return result
 
 is_admin_or_enrolled_in_course = is_admin | is_enrolled_in_course
+is_admin_or_enrolled_in_course_or_load_pending = is_admin_or_enrolled_in_course | course_is_not_loaded
 is_admin_or_enrolled_in_course_id = is_admin | is_enrolled_in_course_id
 is_admin_or_instructor_in_course = is_admin | is_instructor_in_course
 is_admin_or_instructor_in_course_id = is_admin | is_instructor_in_course_id
 
 # api
-rules.add_perm('dashboard.get_course_info', is_admin_or_enrolled_in_course)
```

Comment:
if student is not part of the course first thing canvas denies access /course/12334/external_tool/12233. I got confused since I was access as me ( Admin)

Reply:
Did you tested the case in standalone like this

1. student login and access HIS course 11111
2. After that he try to access course 22222 WHICH HE IS NOT PART OF and course data is loaded for the course 22222. does the code gives access to course or shows permission denied?

Reply:
> Where is the check that is implemented if student is not part of the course and try to log in to course 12334 that he is not part? A student with an URL `/course/12334/external_tool/12233` can access any course since LTI tool id/external_tools for MYLA is same across all the courses.
> 
> Line 68 exactly doing that it was authorizing the user if part of the course or not
The new line 76

Reply:
> Did you tested the case in standalone like this
> 
> 1. student login and access HIS course 11111
> 2. After that he try to access course 22222 WHICH HE IS NOT PART OF and course data is loaded for the course 22222. does the code gives access to course or shows permission denied?

It says "no access or course being loaded" but does not show the image.  This is the one oddball i have seen. It only happens in standalone.

Reply:
That's a permission denied error. is the course data loaded? if YES then it is doing it correctly. if not then we need to investigate. But is shouldn't the way the code works

Reply:
The data is loaded for the course that gives that message for someone not enrolled in the course.